src/ray/util/filesystem.cc

```diff
@@ -0,0 +1,51 @@
+#include "ray/util/filesystem.h"
+
+#include <stdlib.h>
+
+#include "ray/util/logging.h"
+
+#if __cplusplus >= 201703L
+#include <filesystem>
+#elif defined(_WIN32)
+#include <Windows.h>
+#endif
+
+namespace ray {
+
+std::string get_ray_temp_dir() { return join_paths(get_user_temp_dir(), "ray"); }
+
+std::string get_user_temp_dir() {
+  std::string result;
+#if defined(__APPLE__) || defined(__linux__)
+  // Prefer the hard-coded path for now, for compatibility.
+  result = "/tmp";
+#elif __cplusplus >= 201703L
+  result = std::filesystem::temp_directory_path();
```

Comment:
Why don't we remove this altogether for now? It is probably not very portable :D